src/app/replayer/replayer.ml

```diff
@@ -900,12 +884,25 @@ let main ~input_file ~output_file ~archive_uri ~continue_on_error () =
                slot %Ld and sequence number %d"
               ic.global_slot ic.sequence_no ()
       in
+      let%bind unparented_ids =
+        query_db pool
+          ~f:(fun db -> Sql.Block.get_unparented db ())
+          ~item:"unparented ids"
+      in
+      let genesis_block_id =
+        match unparented_ids with
+        | [id] ->
+            id
+        | _ ->
+            failwith "Expected only the genesis block to have an unparented id"
```

Comment:
Might it make sense to list them here, for debugging purposes?

Reply:
I think we may also have orphaned blocks on forks; we don't worry about recovering them elsewhere, but this will break the tool if there are missing blocks anywhere in the db, right?

Perhaps it would be safer to show a warning here?

Reply:
instead, I should filter the list of unparented ids on membership in the `block_ids`; that should yield a singleton